File: bookshelf/src/mfqbxssc/java/com.dingyue.bookshelf/BookShelfItemHolder.java
package com.dingyue.bookshelf;

import android.content.res.Resources;
import android.text.TextUtils;
import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;

import com.bumptech.glide.Glide;
import com.bumptech.glide.load.engine.DiskCacheStrategy;

import net.lzbook.kit.constants.ReplaceConstants;
import net.lzbook.kit.data.bean.Book;
import net.lzbook.kit.utils.AppUtils;
import net.lzbook.kit.utils.Tools;

/**
 * Created by Administrator on 2017/4/13 0013.
 */

public class BookShelfItemHolder extends AbsRecyclerViewHolder<Book> {

    private final ImageView mCheck;
    private final ImageView mImage;
    private final TextView mName, mUpdateTime;
    private final View mFinish, mUpdate;

    public BookShelfItemHolder(View itemView,
                               BookShelfReAdapter.ShelfItemClickListener shelfItemClickListener,
                               BookShelfReAdapter.ShelfItemLongClickListener shelfItemLongClickListener) {
        super(itemView, shelfItemClickListener, shelfItemLongClickListener);

        mCheck = (ImageView) itemView.findViewById(R.id.check_delete);
        mImage = (ImageView) itemView.findViewById(R.id.book_shelf_image);

        mFinish = itemView.findViewById(R.id.book_shelf_status_finish);
        mUpdate = itemView.findViewById(R.id.book_shelf_status_update);
        mName = (TextView) itemView.findViewById(R.id.book_shelf_name);
        mUpdateTime = (TextView) itemView.findViewById(R.id.book_shelf_update_time);
//        mUnread = (TextView) itemView.findViewById(R.id.book_shelf_unread);

    }

    @Override
    public void onBindData(int position, Book book, boolean update, boolean isRemoveMode, boolean
            removeMark) {
        if (!TextUtils.isEmpty(book.name))
            this.mName.setText(book.name);

        if (book.sequence + 1 > book.chapter_count) {
            book.sequence = book.chapter_count - 1;
        }
        int update_count = book.chapter_count - (book.sequence + 1);
//        if(update_count == 0){
//            this.mUnread.setVisibility(View.GONE);
//        }else{
//            this.mUnread.setVisibility(View.VISIBLE);
//            this.mUnread.setText(update_count + "章未读");
//        }

        // 是否连载
        if (((Book) book).status == 2) {
            this.mFinish.setVisibility(View.VISIBLE);
        } else {
            this.mFinish.setVisibility(View.GONE);
        }
        // 是否有更新
        if (!update) {
            this.mUpdate.setVisibility(View.GONE);
        } else {
            this.mUpdate.setVisibility(View.VISIBLE);
            this.mFinish.setVisibility(View.GONE);
        }
        if (this.mUpdateTime != null) {
            this.mUpdateTime.setText(Tools.compareTime(AppUtils.formatter, book
                    .last_updatetime_native) + "更新");
        }

        if (!TextUtils.isEmpty(book.img_url) && !book.img_url.equals(ReplaceConstants.getReplaceConstants().DEFAULT_IMAGE_URL)) {
            Glide.with(itemView.getContext()).load(book.img_url).placeholder(R.drawable.icon_book_cover_default).error((R.drawable.icon_book_cover_default)).diskCacheStrategy(DiskCacheStrategy.ALL).into(this.mImage);
        } else {
            Glide.with(itemView.getContext()).load(R.drawable.icon_book_cover_default).into(this.mImage);
        }

        if (isRemoveMode) {
            this.mCheck.setVisibility(View.VISIBLE);
            this.mFinish.setVisibility(View.GONE);
            this.mUpdate.setVisibility(View.GONE);
            int typeColor = 0;
            Resources.Theme theme = itemView.getContext().getTheme();
            if (removeMark) {
                typeColor = R.drawable.edit_bookshelf_selected;
            } else {
                typeColor = R.drawable.edit_bookshelf_unselected;
            }
            this.mCheck.setBackgroundResource(typeColor);
        } else {
            this.mCheck.setVisibility(View.GONE);
        }
    }
}

File: bookshelf/src/mfqbxssc/java/com.dingyue.bookshelf/BookShelfItemHolder.kt
package com.dingyue.bookshelf

import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.bumptech.glide.Glide
import com.bumptech.glide.load.engine.DiskCacheStrategy
import kotlinx.android.synthetic.mfqbxssc.item_bookshelf_book.view.*
import net.lzbook.kit.app.BaseBookApplication
import net.lzbook.kit.constants.Constants
import net.lzbook.kit.constants.ReplaceConstants
import net.lzbook.kit.data.bean.Book
import net.lzbook.kit.repair_books.RepairHelp

/**
 * Desc 书架页Item
 * Author qiantao
 * Mail tao_qian@dingyuegroup.cn
 * Date 2018/5/12 0002 14:19
 */
class BookShelfItemHolder(parent: ViewGroup) : RecyclerView.ViewHolder(
        LayoutInflater.from(parent.context).inflate(R.layout.item_bookshelf_book, parent, false)) {

    fun bind(book: Book, bookshelfItemListener: BookShelfAdapter.BookShelfItemListener,
             contains: Boolean, remove: Boolean) = with(itemView) {

        if (book.name.isNotEmpty()) txt_book_name.text = book.name
        val sp = BaseBookApplication.getGlobalContext().getSharedPreferences(Constants.SHAREDPREFERENCES_KEY, 0)
        if (RepairHelp.isShowFixBtn(context, book.book_id) && sp.getBoolean(book.book_id, true)) {
            img_book_status.visibility = View.VISIBLE
            img_book_status.setImageResource(R.drawable.bookshelf_item_book_update_icon)
        }else{
            when {
                book.update_status == 1 -> { //更新
                    img_book_status.visibility = View.VISIBLE
                    img_book_status.setImageResource(R.drawable.bookshelf_item_book_update_icon)
                }
                book.status == 2 -> { //完结
                    img_book_status.visibility = View.VISIBLE
                    img_book_status.setImageResource(R.drawable.bookshelf_item_book_finish_icon)
                }
                else -> img_book_status.visibility = View.GONE
            }
        }


        if (book.img_url.isNotEmpty() && book.img_url != ReplaceConstants.getReplaceConstants().DEFAULT_IMAGE_URL) {
            Glide.with(itemView.context)
                    .load(book.img_url)
                    .placeholder(R.drawable.common_book_cover_default_icon)
                    .error(R.drawable.common_book_cover_default_icon)
                    .diskCacheStrategy(DiskCacheStrategy.ALL)
                    .into(img_book_cover)
        } else {
            Glide.with(itemView.context)
                    .load(R.drawable.common_book_cover_default_icon)
                    .into(img_book_cover)
        }

        if (remove) {
            img_book_select_state.visibility = View.VISIBLE
            if (contains) {
                img_book_select_state.setImageResource(R.drawable.bookshelf_item_checked_icon)
            } else {
                img_book_select_state.setImageResource(R.drawable.bookshelf_item_check_icon)
            }
        } else {
            img_book_select_state.visibility = View.GONE
        }

        rl_book_content.setOnClickListener {
            bookshelfItemListener.clickedBookShelfItem(book, adapterPosition)
        }

        rl_book_content.setOnLongClickListener {
            bookshelfItemListener.longClickedBookShelfItem()
        }

    }
}
